client/context/query.go

```diff
@@ -281,6 +305,50 @@ func (ctx CLIContext) ensureBroadcastTx(txBytes []byte) error {
 	return nil
 }
 
+// proofVerify perform response proof verification
+func (ctx CLIContext) proofVerify(path string, resp abci.ResponseQuery) error {
+	// Data from trusted node or subspace query doesn't need verification
+	if ctx.TrustNode || !isQueryStoreWithProof(path) {
```

Comment:
I think the `!isQueryStoreWithProof(path)` should be moved outside to the caller. This provides a better abstraction.